### PR TITLE
Primevue pagination

### PR DESCRIPTION
**Pull request by ilyapozh**
Pagination with primevue and vuex.

### CODE REVIEW

File: my-vue3-app/src/store/reports/actions.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import api from '@/services/api';
+import { Pagination } from './state';
+import { SET_DATA, SET_PAGINATION } from './mutations';
+
+export const FETCH_REPORTS = 'load_reports';
+
+export default {
+  async [FETCH_REPORTS]({ commit }: any, payload: Pagination) {
+    console.log(payload);
+    const data = await api.fetchReports(
```
Comment (tomfun) [inline my-vue3-app/src/store/reports/actions.ts:10]: there is no loader - bad for user experience

File: my-vue3-app/src/store/reports/state.ts [hunk 2]
```diff
@@ -10,15 +10,23 @@ export interface Report {
...
     pageSize: number;
     itemsTotal: number;
-  };
 }
 
 export default {
-  [REPORTS]: [],
+  [REPORTS]: [
+    {
+      title: 'Title',
```
Comment (tomfun) [inline my-vue3-app/src/store/reports/state.ts:25]: remove test data

**Review by tomfun — CHANGES_REQUESTED** [review]

File: my-vue3-app/src/store/reports/state.ts (hunk 2 above)
Comment (ilyapozh) [reply my-vue3-app/src/store/reports/state.ts:25]: Should there be some initial data for first report load?

File: my-vue3-app/package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "primeicons": "^4.1.0",
     "primevue": "^3.8.0",
     "vue": "^3.0.0",
+    "vue-loading-template": "^1.3.2",
```
Comment (tomfun) [inline my-vue3-app/package.json:14]: del

File: my-vue3-app/src/components/ReportList.vue [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
+        mode="indeterminate"
+        style="height: .5em"
+      />
       <SingleReport
         v-for="item in reports"
         v-bind:key="item.id"
         v-bind:item="item"
       ></SingleReport>
     </ul>
+     <vue-loading
```
Comment (tomfun) [inline my-vue3-app/src/components/ReportList.vue:15]: del

File: my-vue3-app/src/components/ReportList.vue [hunk 5]
```diff
@@ -33,19 +40,25 @@ export default defineComponent({
...
       return this.$store.state[REPORTS_MODULE].pagination;
     },
+    reports() {
+      return this.$store.state[REPORTS_MODULE].data;
+    },
   },
   data() {
     return {
-      reports: [],
+      barStatus: false, // todo: @ilyapozh move me to $store
```
Comment (tomfun) [inline my-vue3-app/src/components/ReportList.vue:49]: todo

**Review by tomfun — CHANGES_REQUESTED** [review]

File: my-vue3-app/src/store/reports/mutations.ts [hunk 6]
```diff
@@ -11,4 +14,10 @@ export default {
   [SET_DATA](state: State, data: Array<Report>) {
     state.data = data;
   },
+
+  [SET_LOADING](state: State, isLoading: IsLoading) {
+    console.log(isLoading);
```
Comment (tomfun) [inline my-vue3-app/src/store/reports/mutations.ts:19]: linter

**Review by tomfun — CHANGES_REQUESTED** [review]

**Pull request merged**